salt/_states/metalk8s_checks.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+"""
+State module for handling MetalK8s checks.
+"""
+
+from salt.exceptions import CheckError
+
+__virtualname__ = "metalk8s_checks"
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def module_run(name, **kwargs):
```

Comment:
Why wrap `module.run`? Since we have a custom state module and the corresponding execution module, why not just call it directly?

Something like:

```sls
Check sysctl overrides are properly applied:
  metalk8s_checks.sysctl:
    - params: {{ kubeadm_preflight.mandatory.sysctl_values | json }}
    - require:
      - cmd: Apply sysctl configuration files
```

And:

```python
def sysctl(name, params, **kwargs):
  ret = {
    "name": name,
    "changes": {},
    "result": True,
    "comment": "All sysctl settings are correctly configured."
  }
  error_msg = __salt__["metalk8s_checks.sysctl"](params, raises=False)
  if error_msg:
    ret["result"] = False
    ret["comment"] = error_msg
  return ret
```

Reply:
It's what I did first, but it was to have something more generic.

Reply:
Oh. I see the idea, just wondering what benefit we have (outside of less YAML to write):
- if one module from the list fails (raises a `CheckError`), we'll only get output for this one
- the state ID for the `module.run` will not be specific, so we'll need to rely on the comment to understand what was being checked and what failed

If we really want something able to handle multiple checks (not sure it's worth the effort, TBH), we'll need to do a bit more than just wrapping `module.run` and catching the first `CheckError` that comes our way.
If we really want something able to handle any check, then maybe we don't need to wrap `module.run`, simply:

```python
def run(name, mod, **kwargs):
    ret = {
        "name": name,
        "changes": {},
        "result": True,
        "comment": "Check succeeded.",
    }

    try:
        comment = __salt__[mod](**kwargs)
    except KeyError:
        ret["result"] = False
        ret["comment"] = "Module `{}` does not exist.".format(mod)
    except CheckError as exc:
        ret["result"] = False
        ret["comment"] = str(exc)

    if comment:
        ret["comment"] = comment

    return ret
```

Reply:
And if we really want to ensure this only runs known checks, we can restrict it with:

```python
def run(name, func, **kwargs):
    ret = { ... }

    try:
        comment = __salt__['metalk8s_checks.{}'.format(func)](**kwargs)
    except ...
```

Reply:
Even if we only do one call per module.run at least here we have only one custom state to do all checks instead of writing a states + a module for each check we want to do. 
And btw a state for check does not really make sense at the end, a state should do some modification on the system and not only check stuff, we could imagine in the future having a "tool" that basically just run the module directly.

But no opinion about call state `module.run` vs call our module directly. Btw if we stay will `module.run` at least we can consider this state work the same as all other "classic module.run" states but it's not really an argument because any our states will be really simple so .... as you want

Reply:
done